udata_hydra/analysis/resource.py

```diff
@@ -32,7 +33,7 @@ class Change(Enum):
 
 
 async def analyse_resource(
-    check_id: int, is_first_check: bool, force_analysis: bool = False
+    check_id: int, last_check: dict | None, force_analysis: bool = False
```

Comment:
Since we're using last_check here, we could actually forward it to the different detect change function that fetch it.
However, since it's already quite a PR, let's do that later on

Reply:
I thought about the same, but thought also we should do that later. There are several places in the code where we could avoid re-requesting the check from the DB, but instead passing the record (or the dict) from one method to another.
Let's optimize that later indeed.